http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/DiskV2ServiceInterp.scala

```diff
@@ -147,3 +150,10 @@ case class DiskWithoutWorkspaceException(id: DiskId, traceId: TraceId)
       StatusCodes.Conflict,
       traceId = Some(traceId)
     )
+
+case class DiskWithoutWsmResourceIdException(id: DiskId, traceId: TraceId)
```

Comment:
I am also kind of curious about this ... 
If there is no WsmResource for this, should we clean-up our own state? Mark it or ask the caller to confirm that the resource has been removed?

Reply:
Hmm I think you're correct in this case, the wsmResourceId is set in back leo after the call to WSM to create the disk, so if there's no resourceId, then the resource was never created in WSM so we can just set it as deleted in Leo.